src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -977,10 +961,16 @@ public void SetCookies(Uri uri, string cookieHeader)
         }
     }
 
-    internal sealed class PathList
+    internal struct PathList
     {
-        private readonly SortedList<string, CookieCollection> _list = new SortedList<string, CookieCollection>(PathListComparer.StaticInstance);
-        private readonly object _lock = new object();
+        private readonly SortedList<string, CookieCollection> _list;
```

Comment:
Consider adding a comment here how usage of PathList depends on it being shallowly immutable, and adding any mutable fields to it would result in breaks.